components/server/src/workspace/workspace-starter.ts

```diff
@@ -720,28 +719,39 @@ export class WorkspaceStarter {
             await this.messageBus.notifyOnInstanceUpdate(workspace.ownerId, instance);
 
             // If we just attempted to start a workspace for a prebuild - and that failed, we have to fail the prebuild itself.
+            await this.failPrebuildWorkspace({ span }, err, workspace);
+        } catch (err) {
+            TraceContext.setError({ span }, err);
+            log.error(
+                { workspaceId: workspace.id, instanceId: instance.id, userId: workspace.ownerId },
+                "cannot properly fail workspace instance during start",
+                err,
+            );
+        } finally {
+            span.finish();
+        }
+    }
+
+    protected async failPrebuildWorkspace(ctx: TraceContext, err: Error, workspace: Workspace) {
+        const span = TraceContext.startSpan("failInstanceStart", ctx);
+        try {
             if (workspace.type === "prebuild") {
                 const prebuild = await this.workspaceDb.trace({ span }).findPrebuildByWorkspaceID(workspace.id);
                 if (prebuild && prebuild.state !== "failed") {
                     prebuild.state = "failed";
                     prebuild.error = err.toString();
 
                     await this.workspaceDb.trace({ span }).storePrebuiltWorkspace(prebuild);
-                    await this.messageBus.notifyHeadlessUpdate({ span }, workspace.ownerId, workspace.id, <
-                        HeadlessWorkspaceEvent
-                    >{
+                    await this.messageBus.notifyHeadlessUpdate({ span }, workspace.ownerId, workspace.id, {
                         type: HeadlessWorkspaceEventType.Failed,
-                        // TODO: `workspaceID: workspace.id` not needed here? (found in ee/src/prebuilds/prebuild-queue-maintainer.ts and ee/src/bridge.ts)
+                        workspaceID: workspace.id, // required at least in PrebuildStatusMaintainer
```

Comment:
todo: search logs for "cannot read workspaceID of undefined", this looks like the reason for failing PrebuildStatusMaintainer calls. worth a small PR